src/main/scala/uk/gov/nationalarchives/tdr/api/service/CustomMetadataPropertiesService.scala

```diff
@@ -88,7 +89,8 @@ class CustomMetadataPropertiesService(customMetadataPropertiesRepository: Custom
       metadataValues.toList,
       fp.uiordinal.getOrElse(Int.MaxValue),
       fp.allowexport,
-      fp.exportordinal.map(_.toInt)
+      fp.exportordinal.map(_.toInt),
+      Set(MandatoryMetadata, SystemProperty).contains(fp.propertygroup.get)
```

Comment:
Can you create a variable for the protected field in class and then use it here?

`val protectedPropertyGroups = Set(MandatoryMetadata, SystemProperty)`